source/ed/data.cpp

```diff
@@ -735,28 +735,54 @@ EdTZWrapper::get_dst_periods(const boost::gregorian::date_period& validity_perio
             //from the previous end date to the beggining of the dst next year
-            res.push_back({ {res.back().period.end(), boost_timezone->dst_local_start_time(year).date()},
-                            boost_timezone->base_utc_offset() });
+            if (is_dst_start_smaller) {
+                res.push_back({ {res.back().period.end(), boost_timezone->dst_local_start_time(year).date()},
+                                boost_timezone->base_utc_offset() });
+            } else {
+                res.push_back({ {res.back().period.end(), boost_timezone->dst_local_end_time(year).date()},
+                                boost_timezone->base_utc_offset() + boost_timezone->dst_offset() });
+            }
+
         } else {
             //for the first elt, we add a non dst
             auto first_day_of_year = boost::gregorian::date(year, 1, 1);
             if (boost_timezone->dst_local_start_time(year).date() != first_day_of_year) {
-                res.push_back({ {first_day_of_year, boost_timezone->dst_local_start_time(year).date()},
-                                boost_timezone->base_utc_offset() });
+                if (is_dst_start_smaller) {
+                    res.push_back({ {first_day_of_year, boost_timezone->dst_local_start_time(year).date()},
+                                    boost_timezone->base_utc_offset() });
+                } else {
+                    res.push_back({ {first_day_of_year, boost_timezone->dst_local_end_time(year).date()},
+                                    boost_timezone->base_utc_offset() + boost_timezone->dst_offset() });
+                }
+
             }
         }
-        res.push_back({ {boost_timezone->dst_local_start_time(year).date(), boost_timezone->dst_local_end_time(year).date()},
-                        boost_timezone->base_utc_offset() + boost_timezone->dst_offset() });
+        if (is_dst_start_smaller){
+            res.push_back({ {boost_timezone->dst_local_start_time(year).date(), boost_timezone->dst_local_end_time(year).date()},
+                            boost_timezone->base_utc_offset() + boost_timezone->dst_offset() });
+        } else {
+            res.push_back({ {boost_timezone->dst_local_end_time(year).date(), boost_timezone->dst_local_start_time(year).date()},
+                            boost_timezone->base_utc_offset() });
+        }
     }
+
     //we add the last non DST period
-    res.push_back({ {res.back().period.end(), boost::gregorian::date(years.back() + 1, 1, 1)},
-                    boost_timezone->base_utc_offset() });
+    if (is_dst_start_smaller) {
+        res.push_back({ {res.back().period.end(), boost::gregorian::date(years.back() + 1, 1, 1)},
+                        boost_timezone->base_utc_offset() });
+    } else {
+        res.push_back({ {res.back().period.end(), boost::gregorian::date(years.back() + 1, 1, 1)},
+                        boost_timezone->base_utc_offset()  + boost_timezone->dst_offset()});
```

Comment:
same here, I'd rather have it stored in a variable (maybe even with a  ternary operator)